### PR TITLE
ability to unbundle xxhash

### PR DESCRIPTION
**Pull request by FelixSchwarz**
This should fix issue #4592 . I tested this (briefly) on Fedora 30 and the test suite passes for me.

### CODE REVIEW

File: src/borg/algorithms/checksums.pyx [hunk 1]
```diff
@@ -80,25 +82,29 @@ def xxh64(data, seed=0):
 
 
 cdef class StreamingXXH64:
-    cdef XXH64_state_t state
+    cdef XXH64_state_t* state
 
     def __cinit__(self, seed=0):
+        self.state = XXH64_createState();
```
Comment (ThomasWaldmann) [inline src/borg/algorithms/checksums.pyx]: no semicolon.

Comment (ThomasWaldmann) [reply src/borg/algorithms/checksums.pyx]: also: can this call go wrong somehow, do we need an error handler?

Comment (FelixSchwarz) [reply src/borg/algorithms/checksums.pyx]: `XXH64_createState()` is basically just a renamed `malloc()` ([source](https://github.com/Cyan4973/xxHash/blob/701423eeda62b151a437444290a2669f46324f13/xxhash.c#L454)).

I could check for `self.state is None` (assuming cython would map `NULL` to `None`) and raise an exception afterwards but I'm not sure that would work in an actual out-of-memory situation.

I can of course implement something like this if you prefer (also other suggestions welcome).

Comment (ThomasWaldmann) [reply src/borg/algorithms/checksums.pyx]: hmm, i had a look at other code parts, we do not check for allocation errors of tiny structs there either. and yes, guess python could then be in a rather dysfunctional state anyway.

so keep as is and just remove the semicolon (and collapse the fixup changeset).

**Comment by FelixSchwarz** [comment]
ok, I force-pushed new commits which do not include the semicolon. Anything I missed?

**Comment by ThomasWaldmann** [comment]
thanks!

**Pull request merged**